temporalio/worker/_tuning.py

```diff
@@ -0,0 +1,185 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Literal, Optional, Union
+
+from typing_extensions import TypeAlias
+
+import temporalio.bridge.worker
+
+
+@dataclass(frozen=True)
+class FixedSizeSlotSupplier:
+    """A fixed-size slot supplier that will never issue more than a fixed number of slots."""
+
+    num_slots: int
+    """The maximum number of slots that can be issued"""
+
+
+@dataclass(frozen=True)
+class ResourceBasedTunerOptions:
```

Comment:
```suggestion
class ResourceBasedTunerConfig:
```

While we try to use kwargs as much as possible, I can understand the use case here, but I think we should suffix it with `Config` (we've done this in a few other places)